### PR TITLE
Optimize the suboptimal image locality algorithm

### PR DESCRIPTION
**Pull request by anfernee**
**What this PR does / why we need it**: Cut loop depth from 3 to 2. 

**Release note**:
```release-note
None
```

### CODE REVIEW

File: plugin/pkg/scheduler/algorithm/priorities/image_locality.go [hunk 1]
```diff
@@ -49,31 +47,34 @@ func ImageLocalityPriorityMap(pod *v1.Pod, meta interface{}, nodeInfo *scheduler
...
 	}
-	// Return which bucket the given size belongs to
-	return score
+
+	return int((int64(schedulerapi.MaxPriority) * (sumSize - minImgSize) / (maxImgSize - minImgSize)) + 1)
 }
 
-// checkContainerImageOnNode checks if a container image is present on a node and returns its size.
-func checkContainerImageOnNode(node *v1.Node, container *v1.Container) int64 {
+func totalImageSize(node *v1.Node, containers []v1.Container) int64 {
```
Comment (bsalamat) [inline plugin/pkg/scheduler/algorithm/priorities/image_locality.go]: Please add a comment before this function about what it does.

File: plugin/pkg/scheduler/algorithm/priorities/image_locality.go [hunk 2]
```diff
@@ -49,31 +47,34 @@ func ImageLocalityPriorityMap(pod *v1.Pod, meta interface{}, nodeInfo *scheduler
 // 1. Split image size range into 10 buckets.
 // 2. Decide the priority of a given sumSize based on which bucket it belongs to.
 func calculateScoreFromSize(sumSize int64) int {
-	var score int
 	switch {
+	// score == 0 means none of the images required by this pod are present on this
```
Comment (bsalamat) [inline plugin/pkg/scheduler/algorithm/priorities/image_locality.go]: I think you should update this comment and replace 'score == 0 means ...' with 'zero means...'. You should also move it right before the return.

**Review by bsalamat — COMMENTED** [review]
Thanks, @anfernee! Looks good. I just have a couple of minor comments.

**Comment by anfernee** [comment]
great. all green now.

**Comment by anfernee** [comment]
thanks @bsalamat

**Comment by bsalamat** [comment]
Thank you, @anfernee for the improvement.

**Pull request merged**